packages/flutter/lib/src/material/bottom_app_bar_theme.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines a theme for [BottomAppBar] widgets.
+///
+/// Descendant widgets obtain the current [BottomAppBarTheme] object using
+/// `BottomAppBarTheme.of(context)`. Instances of [BottomAppBarTheme] can be
+/// customized with [BottomAppBarTheme.copyWith].
+///
+/// See also:
+///
+///  * [BottomAppBar], a material bottom app bar that can be customized using
+///    this [BottomAppBarTheme].
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class BottomAppBarTheme extends Diagnosticable {
+  /// Creates a dialog theme that can be used for [ThemeData.BottomAppBarTheme].
+  const BottomAppBarTheme({
+    this.color,
+    this.elevation,
+    this.shape,
+  });
+
+  /// Default value for [BottomAppBar.color].
+  final Color color;
+
+  /// Default value for [BottomAppBar.elevation].
+  final double elevation;
+
+  /// Default value for [Dialog.shape].
+  final NotchedShape shape;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  BottomAppBarTheme copyWith({
+    Color color,
+    double elevation,
+    NotchedShape shape,
+  }) {
+    return BottomAppBarTheme(
+      color: color ?? this.color,
+      elevation: elevation ?? this.elevation,
+      shape: shape ?? this.shape,
+    );
+  }
+
+  /// The data from the closest [BottomAppBarTheme] instance given the build context.
+  static BottomAppBarTheme of(BuildContext context) {
```

Comment:
Does being static have any effect on whether it can change at runtime?

Reply:
Nope, the method itself is static because it doesn't depend on the individual members of any given instance, but the object it returns is dependent on the `context` parameter. So as Widgets are rebuilt they will pass new `context`s to this method which will fetch the most up to date theme.

Reply:
Duh. That was a stupid question on my part.